Orange/widgets/visualize/owscatterplot.py

```diff
@@ -419,17 +420,36 @@ def set_subset_data(self, subset_data):
 
     # called when all signals are received, so the graph is updated only once
     def handleNewSignals(self):
-        self.graph.new_data(self.data_metas_X, self.subset_data)
+        if not sp.issparse(self.data.X):
+            self.graph.new_data(self.data_metas_X, self.subset_data)
         if self.attribute_selection_list and \
                 all(attr in self.graph.domain
                         for attr in self.attribute_selection_list):
             self.attr_x = self.attribute_selection_list[0]
             self.attr_y = self.attribute_selection_list[1]
         self.attribute_selection_list = None
-        self.update_graph()
+        if not sp.issparse(self.data.X):
+            self.update_graph()
         self.cb_class_density.setEnabled(self.graph.can_draw_density())
         self.cb_reg_line.setEnabled(self.graph.can_draw_regresssion_line())
         self.unconditional_commit()
+        self.corpus_to_table()
+
+    def corpus_to_table(self):
```

Comment:
Could you rename this to something else? In theory we could have sparse table also in Orange, nut just in Text.